pkg/framework/plugins/podlifetime/validation.go

```diff
@@ -44,15 +44,29 @@ func ValidatePodLifeTimeArgs(obj runtime.Object) error {
 		}
 	}
 	podLifeTimeAllowedStates := sets.New(
+		// Pod Status Phase
 		string(v1.PodRunning),
 		string(v1.PodPending),
+		string(v1.PodUnknown),
 
+		// Pod Status Reasons
+		"NodeAffinity",
+		"NodeLost",
+		"Shutdown",
+		"UnexpectedAdmissionError",
```

Comment:
Some (maybe all) of these are status reasons for Pods that have status Failed, which generally has been handled by `RemoveFailedPods` strategy as opposed to `PodLifeTime`.

We have discussed merging the two strategies in the past so I'm not opposed to this idea.

Let's give it a couple of more days to see if other maintainers want to chime in on our approach going forward.

Reply:
Great. I’m willing to join and improve it